www/js/app.js

```diff
@@ -1917,27 +1917,34 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     }
 
     function goToRandomArticle() {
-        $("#searchingArticles").show();
-        selectedArchive.getRandomDirEntry(function(dirEntry) {
-            if (dirEntry === null || dirEntry === undefined) {
-                $("#searchingArticles").hide();
-                uiUtil.systemAlert("Error finding random article", "Error finding article");
-            } else {
-                // We fall back to the old A namespace to support old ZIM files without a text/html MIME type for articles
-                // DEV: If articlePtrPos is defined in zimFile, then we are using a v1 article-only title listing. By definition,
-                // all dirEntries in an article-only listing must be articles.
-                if (selectedArchive._file.articlePtrPos || dirEntry.getMimetype() === 'text/html' || dirEntry.namespace === 'A') {
-                    params.isLandingPage = false;
-                    $('#activeContent').hide();
-                    $('#searchingArticles').show();
-                    readArticle(dirEntry);
+        if (selectedArchive !== null && selectedArchive.isReady()) {
+            $("#searchingArticles").show();
+            selectedArchive.getRandomDirEntry(function(dirEntry) {
```

Comment:
ESLint style suggests adding a space between `function` and the parentheses for anonymous functions. I realize it was originally like this, but is worth correcting.